projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+use Automattic\Jetpack_Boost\Contracts\Is_Always_On;
+use Automattic\Jetpack_Boost\Contracts\Pluggable;
+
+class Page_Cache implements Pluggable, Is_Always_On {
+	/*
+	 * Sets up the advanced-cache.php file and if that works, adds the WP_CACHE
+	 * define to wp-config.php
+	 * These are used by WordPress to load the caching system before most of
+	 * WordPress is loaded.
+	 */
+	public function setup() {
+		if ( is_admin() && isset( $_GET['page'] ) && $_GET['page'] === 'jetpack-boost' ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+			$result = $this->create_advanced_cache();
+			if ( $result === true ) {
+				$result = $this->add_wp_cache_define();
+				if ( $result !== true ) {
+					return $result;
+				}
+			} else {
+				return $result;
+			}
+
+			return true;
+		}
+	}
+
+	/*
+	 * Creates the advanced-cache.php file.
+	 *
+	 * Returns true if the files were setup correctly, or WP_Error if there was a problem.
+	 * @return bool|WP_Error
+	 */
+	private function create_advanced_cache() {
+		global $wp_filesystem;
+
+		$advanced_cache_filename = ABSPATH . 'wp-content/advanced-cache.php';
+		if ( $wp_filesystem->exists( $advanced_cache_filename ) ) {
+			$content = $wp_filesystem->get_contents( $advanced_cache_filename );
+			if ( strpos( $content, 'Boost Cache Plugin 0.1' ) !== false ) {
+				return true;
+			} else {
+				return new \WP_Error( 'advanced-cache.php exists but is not the correct file' );
+			}
+		} else {
+			$contents = '<?php
+// Boost Cache Plugin 0.1
+require_once( ABSPATH . \'/wp-content/plugins/boost/app/modules/cache/Boost_File_Cache.php\' );
+
+( new Automattic\Jetpack_Boost\Modules\Page_Cache\Boost_File_Cache() )->serve();
+';
+
+			$result = $wp_filesystem->put_contents( $advanced_cache_filename, $contents, FS_CHMOD_FILE );
+			if ( $result === false ) {
+				return new \WP_Error( 'Could not write to advanced-cache.php' );
+			}
+		}
+
+		return true;
+	}
```

Comment:
Maybe we could avoid adding the version check in there if we just include a file from Boost, that does the require and `serve()`?

I think we should simplify the contents of advanced-cache.php as much as possible, as we might run into issues if we decide to change up structures or reorganize the file structure.

Reply:
Yeah, the version comment has to go directly in `advanced-cache.php` to have any value here.

As for the rest of it, we could put it in a php file of its own and require that from advanced-cache. That could shorten the advanced-cache from 3 lines to 2. Personally I'm not too fussed either way. :)

Reply:
> Yeah, the version comment has to go directly in advanced-cache.php to have any value here.

Why is that necessary? I'm missing the point of adding a lot of things to the file.

Reply:
@dilirity - Mainly, it's to allow us to update the advanced-cache.php more easily in the future, if an update is necessary.

Reply:
Oh I see, doh :D Compare if it's outdated. Got it.